semantic_conventions/http-common.yaml

```diff
@@ -65,7 +65,6 @@ groups:
 
           SHOULD include the [application root](/specification/trace/semantic_conventions/http.md#http-server-definitions) if there is one.
       - ref: server.address
-        requirement_level: required
```

Comment:
I think this should stay required / conditionally required on the server side. However, the HTTP Host header and request target could be removed from the potential sources, leaving only the "primary server name" if one was matched.

Reply:
Essentially, if we believe that the majority of services have just one valid combination of `server.address` and  `server.port` they respond on, it's a sign we need a resource attribute.

Multi-tenant applications should still use resource attributes.

Reply:
I mean the `service.name` resource attribute.  Within one `service.name`, address and port are mostly static, there is no need to aggregate by them when we already aggregate by the `service.name` and other resource attributes on metrics.

Forming a URL  does not seem like a metric-related task, right?

Reply:
> Also, on traces we still want to capture the host header contents. Inconsistent semantics across signals has been recently debated in semconv/spec meetings (wrt https://github.com/open-telemetry/semantic-conventions/pull/17, https://github.com/open-telemetry/semantic-conventions/pull/35) with the conclusion to keep semantics consistent.

this is the key reason to me why we want to make it opt-in instead of altering the semantics of the attribute for metrics only

Reply:
> Forming a URL does not seem like a metric-related task, right?

Why not? You could show your metrics by URL.
Also, isn't your change affecting traces and logs as well?

Reply:
> Why not? You could show your metrics by URL.

does it make much sense to format URL as a dimension (from individual scheme, host, and port)?

You can as well format it from scheme and service name or a combination of resource attributes.


> Also, isn't your change affecting traces and logs as well?

address and port, as you can see in this PR have `recommended` level on traces. We do not have any HTTP semconv for logs, so there is nothing about them.